src/vs/workbench/contrib/positronPreviewEditor/browser/positronPreviewEditor.tsx

```diff
@@ -0,0 +1,177 @@
+}
+
+export class PositronPreviewEditor
+	extends EditorPane
+	implements IReactComponentContainer {
+	private readonly _container: HTMLElement;
+
+	private _positronReactRenderer?: PositronReactRenderer;
+
+	private _width = 0;
+
+	private _height = 0;
+
+	private _identifier?: string;
+
+	private readonly _onSizeChangedEmitter = this._register(new Emitter<ISize>());
+
+	private readonly _onVisibilityChangedEmitter = this._register(
+		new Emitter<boolean>()
+	);
+
+	private readonly _onSaveScrollPositionEmitter = this._register(
+		new Emitter<void>()
+	);
+
+	private readonly _onRestoreScrollPositionEmitter = this._register(
+		new Emitter<void>()
+	);
+
+	private readonly _onFocusedEmitter = this._register(new Emitter<void>());
+
+	get identifier(): string | undefined {
+		return this._identifier;
+	}
+
+	get width() {
+		return this._width;
+	}
+
+	get height() {
+		return this._height;
+	}
+
+	get containerVisible() {
+		return this.isVisible();
+	}
+
+	takeFocus(): void {
+		this.focus();
+	}
+
+	readonly onSizeChanged = this._onSizeChangedEmitter.event;
+
+	readonly onVisibilityChanged: Event<boolean> =
+		this._onVisibilityChangedEmitter.event;
+
+	readonly onSaveScrollPosition: Event<void> =
+		this._onSaveScrollPositionEmitter.event;
+
+	readonly onRestoreScrollPosition: Event<void> =
+		this._onRestoreScrollPositionEmitter.event;
+
+	readonly onFocused: Event<void> = this._onFocusedEmitter.event;
+
+	constructor(
+		readonly _group: IEditorGroup,
+		@IPositronPreviewService
+		private readonly _positronPreviewService: IPositronPreviewService,
+		@IStorageService storageService: IStorageService,
+		@ITelemetryService telemetryService: ITelemetryService,
+		@IThemeService themeService: IThemeService
+	) {
+		super(
+			PositronPreviewEditorInput.EditorID,
+			_group,
+			telemetryService,
+			themeService,
+			storageService
+		);
+
+		this._container = DOM.$('.positron-preview-editor-container');
+	}
+
+	private renderContainer(): void {
+		if (!this._positronReactRenderer) {
+			this._positronReactRenderer = new PositronReactRenderer(this._container);
+		}
+
+		this._positronReactRenderer.render(
+			<PositronPreviewContextProvider
+				positronPreviewService={this._positronPreviewService}
+			>
+				<EditorPreviewContainer
+					preview={this._positronPreviewService.activePreviewWebview}
+					visible={this.containerVisible}
+					width={this._width}
+					height={this._height}
+				/>
+			</PositronPreviewContextProvider>
+		);
+	}
+
+	private disposeReactRenderer(): void {
+		this._positronReactRenderer?.dispose();
+		this._positronReactRenderer = undefined;
+	}
+
+	protected override createEditor(parent: HTMLElement): void {
+		parent.appendChild(this._container);
+	}
+
+	override async setInput(
+		input: PositronPreviewEditorInput,
+		options: IPositronPreviewEditorOptions,
+		context: IEditorOpenContext,
+		token: CancellationToken
+	): Promise<void> {
+		this.renderContainer();
+		this.onSizeChanged((event: ISize) => {
```

Comment:
This just bounces back and forth between here and [the `PositronPreviewViewPane` redraw](https://github.com/posit-dev/positron/blob/b9301b6ee98046f19a4a797ee15d6b51607fde52/src/vs/workbench/contrib/positronPreview/browser/positronPreviewView.tsx#L240), so the content will hop between Viewer and Editor pane depending on where you resize 😩